text/0074/0074-entity-slicing.md

```diff
@@ -0,0 +1,566 @@
+3. **Flexible**- Entity slicing should work on a variety of data storage formats.
+
+
+
+## Example
+
+Here is a schema involving users and documents:
+
+```
+entity User in [User];
+
+entity Metadata = {
+   owner: User,
+   time: String,
+};
+
+entity Document = {
+  metadata: Metadata,
+  readers: Set<User>,
+};
+
+action Read, Edit appliesTo {
+  principal: [User],
+  resource: [Document],
+};
+```
+
+
+Here are a couple Cedar policies conforming to the schema:
+
+```
+// allow documents to be read by anyone in the readers set
+permit(
+  principal,
+  action == Action::"Read",
+  resource
+)
+when
+{
+  resource.readers.contains(principal)
+};
+
+// allow documents to be read and edited by owners
+permit(
+  principal,
+  action in [Action::"Read", Action::"Edit"],
+  resource
+)
+when
+{
+  resource.metadata.owner == principal
+};
+
+// allow read if the user is a global admin
+permit(
+  principal in User::"GlobalAdmin",
+  action in [Action::"Read"],
+  resource
+);
+```
+
+Now, we would like to answer the question: given a request that a **User** would like to **Read** a **Document,** what data is required?
+
+To answer the request correctly, Cedar requires two data paths and one entity’s ancestors in the entity store:
+
+```
+Request type: (User, Read, Document)
+Data required:
+   resource.readers
+   resource.metadata.owner
+   principal.**ancestors** (where ancestors are the ancestors in the entity hierarchy)
+```
+
+On the other hand, if the **User** would like to **Edit** a **Document**, we only need one piece of data:
+
+```
+Request type: (User, Edit, Document)
+Data required:
+  resource.metadata.owner
+```
+
+
+
+## Example Usage
+
+![A client communicates with a Cedar Server](clientserver.png)
+
+The image above shows an example usage of the entity manifest. On the left, a client has access to the entity data and a cached entity manifest. On the right, a server stores the schema and policies.
+
+When a client would like to answer a Cedar `Request`, it first consults the cached entity manifest to load the entity slice. Then, it sends the entity slice and the request to the server. The server returns the authorization result.
+
+However, the server or client must take care to ensure that the client’s entity manifest is up-to-date. Since the entity manifest is computed based on the schema and policies, is must be re-computed whenever the schema or policies change. There are multiple ways to ensure that the cache is up to date, one being to tag each entity manifest uniquely and check on each request.
+
+Looking to the future, we hope to provide several options and guidance for how to use the entity manifest:
+
+1. Use the `SimplifiedEntityLoader` api (see later section).
+2. Use a specialized entity loader when the format of the database sufficiently matches the schema.
+3. Write custom entity loading that leverages the entity manifest.
```

Comment:
How is option 2 different from option 3?

Reply:
Option 2 is to use a specialized loader that Cedar provides. For example, if we provided a loader that interfaced with SQL databases, it would be super convenient and performant.

Option 3 is for the user to write the specialized loader.